common/exception/class.MethodNotAllowed.php

```diff
@@ -25,8 +25,32 @@
  */
 class common_exception_MethodNotAllowed extends common_exception_BadRequest
 {
+    /**
+     * @var string[]|null
+     */
+    protected $allowedMethods;
+
+    /**
+     * @param string|null $message
+     * @param int $code
+     * @param string[]|null $allowedMethods
+     */
+    public function __construct($message = null, int $code = 0, $allowedMethods = null)
```

Comment:
Did we finally moved to php 7 (`int` type hint)?
What if restrict last parameter type to `array` and set default value `[]` ?

Reply:
Thx, int type hint removed.

> What if restrict last parameter type to array and set default value [] ?

I have thought about it too, but decided to keep `string[]|null` to explicitly express that in case of `null` allowed methods are not specified by code which threw the exception in opposite to case when we have empty array which means empty list of allowed methods. Of course, the proper code shouldn't pass an empty array.

Reply:
But why not restrict type to array? You still can use null default value, but when it passed, it should be an array:
```
public function __construct($message = null, int $code = 0, array $allowedMethods = null)
```

Reply:
You are right, fixed